### PR TITLE
update edit tests, fix checkstyle and bugs

### PR DESCRIPTION
**Pull request by tanveersingh10**

### CODE REVIEW

File: src/test/java/seedu/address/logic/commands/EditCommandTest.java [hunk 1]
```diff
@@ -84,75 +87,57 @@ public void execute_noFieldSpecifiedUnfilteredList_success() {
...
         EditPersonDescriptor descriptor = new EditPersonDescriptorBuilder(firstPerson).build();
-        EditCommand editCommand = new EditCommand(INDEX_SECOND_PERSON, descriptor);
+        Ic nricOfSecondPerson = model.getFilteredPersonList().get(1).getIc();
+        EditCommand editCommand = new EditCommand(nricOfSecondPerson, descriptor);
 
         assertCommandFailure(editCommand, model, EditCommand.MESSAGE_DUPLICATE_PERSON);
     }
 
     @Test
-    public void execute_duplicatePersonFilteredList_failure() {
```
Comment (cmHuang777) [inline src/test/java/seedu/address/logic/commands/EditCommandTest.java:112]: Are there any tests to replace editing other persons into duplicates?

**Review by cmHuang777 — CHANGES_REQUESTED** [review]
Overall looks good, just some nits on a test case deleted.

File: src/main/java/seedu/address/logic/commands/EditCommand.java [hunk 2]
```diff
@@ -81,12 +90,33 @@ public CommandResult execute(Model model) throws CommandException {
...
 
-        Person personToEdit = lastShownList.get(index.getZeroBased());
-        Person editedPerson = createEditedPerson(personToEdit, editPersonDescriptor);
+        //developer assumption - can't have 2 people with same IC
+        assert personToEditList.size() < 2;
+
+        Person personToEdit = personToEditList.get(0);
+        Person editedPerson;
+
+        if (personToEdit instanceof Patient) {
```
Comment (chonguschonguschongus) [inline src/main/java/seedu/address/logic/commands/EditCommand.java:107]: Possible to extract out this part of the function? Think the execute is a bit too long and will cost us readability.

**Review by chonguschonguschongus — COMMENTED** [review]
LGTM. Overall code quality is great except the slightly longer execute() method for EditCommand.

**Review by cmHuang777 — APPROVED** [review]

**Review by cmHuang777 — APPROVED** [review]
We will merge first then fix the tests in a separate pr

**Review by Mohammed-Faizzzz — APPROVED** [review]
We'll merge this first, and introduce more testcases ASAP.

**Pull request merged**